### ISSUE TITLE
server crash

### ISSUE THREAD

**Issue by askovpen** [L-EARN/NoCheatZ-4#61]
```
PutClientInServer: no info_player_start on level
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, VModEnable 1)
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, vban 0 0)
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, joingame )
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, jointeam 0 1)
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : VirtualTableHook : function 0xF2DE29F0 at 0xF349D544 replaced by 0xEC7E1D32.
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : VirtualTableHook : function 0xF2DE29F0 at 0xF348EE44 replaced by 0xEC7E1D32.
L 07/05/2016 - 14:11:30: [csgo_country_v2.smx] 1 round end, 1:0(3) - 16 - $2000$
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, jointeam 3 0)
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : VirtualTableHook : function 0xF2DE29F0 at 0xF348F5E4 replaced by 0xEC7E1D32.
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : VirtualTableHook : function 0xF2DE29F0 at 0xF348FD84 replaced by 0xEC7E1D32.
Error: CreateBot() failed.
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : VirtualTableHook : function 0xF2DE29F0 at 0xF348A9A4 replaced by 0xEC7E1D32.
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] Starting to record the game in NoCheatZ-autorecords/$2000$-07-05-16_14-11-33.dem
Recording GOTV demo to NoCheatZ-autorecords/$2000$-07-05-16_14-11-33.dem...
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : VirtualTableHook : function 0xF2DE29F0 at 0xF349CAC4 replaced by 0xEC7E1D32.
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, +lookatweapon 16)
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, -lookatweapon 16)
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, drop )
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, drop )
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, drop )
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, drop )
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, joingame )
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, jointeam 0 1)
L 07/05/2016 - 14:12:02: [csgo_country_v2.smx] 2 round end, 2:0(3) - 8 - $2000$
srcds_linux: /home/steam/d/NoCheatZ-4/server-plugin/Code/Systems/Blockers/WallhackBlocker.h:102: bool VisCache::IsValid(const NczPlayer*, const NczPlayer*) const: Assertion `pa && pb' failed.

Program received signal SIGABRT, Aborted.
0xf7fda9f2 in ?? () from /lib/ld-linux.so.2
(gdb) bt
#0  0xf7fda9f2 in ?? () from /lib/ld-linux.so.2
#1  0xf7dc6977 in raise () from /lib32/libc.so.6
#2  0xf7dc7f49 in abort () from /lib32/libc.so.6
#3  0xf7dbfa27 in ?? () from /lib32/libc.so.6
#4  0xf7dbfaab in __assert_fail () from /lib32/libc.so.6
#5  0xec7f19ef in VisCache::IsValid (this=0x9006740, pa=0xb137300, pb=0x0)
    at /home/steam/d/NoCheatZ-4/server-plugin/Code/Systems/Blockers/WallhackBlocker.h:102
#6  0xec7f018e in WallhackBlocker::RT_SetTransmitCallback (this=0x9003ab0, sender_player=..., receiver_player=...)
    at /home/steam/d/NoCheatZ-4/server-plugin/Code/Systems/Blockers/WallhackBlocker.cpp:156
#7  0xec7e1e57 in SetTransmitHookListener::RT_nSetTransmit (This=0xc1bce78, pInfo=0xbca66c0, bAlways=false)
    at /home/steam/d/NoCheatZ-4/server-plugin/Code/Hooks/SetTransmitHookListener.cpp:70
#8  0xf2f81ee0 in CServerGameEnts::CheckTransmit(CCheckTransmitInfo*, unsigned short const*, int) ()
   from /mnt/p/home/steam/csgo/csgo/bin/server.so
#9  0xf69b771b in SV_ComputeClientPacks(int, CGameClient**, CFrameSnapshot*) ()
   from /mnt/p/home/steam/csgo/bin/engine.so
#10 0xf69b402e in CGameServer::SendClientMessages(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#11 0xf69b43b1 in SV_SendClientUpdates(bool, bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#12 0xf69b44e0 in SV_Frame(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#13 0xf692308e in _Host_RunFrame_Server(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#14 0xf6923d2c in _Host_RunFrame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#15 0xf6924468 in Host_RunFrame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#16 0xf692d714 in CHostState::State_Run(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#17 0xf692d9e6 in CHostState::FrameUpdate(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#18 0xf692daf9 in HostState_Frame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#19 0xf69ca66a in CEngine::Frame() () from /mnt/p/home/steam/csgo/bin/engine.so
#20 0xf69c75f6 in CDedicatedServerAPI::RunFrame() () from /mnt/p/home/steam/csgo/bin/engine.so
#21 0xf784cc2d in RunServerIteration(bool) () from ./bin/dedicated.so
#22 0xf784ccb8 in RunServer(bool) () from ./bin/dedicated.so
#23 0xf69c76ed in CModAppSystemGroup::Main() () from /mnt/p/home/steam/csgo/bin/engine.so
#24 0xf6b17e70 in CAppSystemGroup::Run() () from /mnt/p/home/steam/csgo/bin/engine.so
#25 0xf69c88b7 in CDedicatedServerAPI::ModInit(ModInfo_t&) () from /mnt/p/home/steam/csgo/bin/engine.so
#26 0xf784cd94 in CDedicatedAppSystemGroup::Main() () from ./bin/dedicated.so
#27 0xf7870740 in CAppSystemGroup::Run() () from ./bin/dedicated.so
#28 0xf787196e in CSteamApplication::Main() () from ./bin/dedicated.so
#29 0xf7870740 in CAppSystemGroup::Run() () from ./bin/dedicated.so
#30 0xf7812d73 in main () from ./bin/dedicated.so
#31 0x080486f9 in main ()

**Comment by askovpen** [comment]
```
Client "Lamar" connected (109.165.90.96:27005).
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientConnect (AllowConnect: Yes, EE0BB050 -> , Lamar, 109.165.90.96:27005, Connection rejected by game
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE0BB060 -> player, joingame )
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE0BB060 -> player, jointeam 0 1)
L 07/05/2016 - 14:38:06: [csgo_country_v2.smx] 4 round end, 1:3(4) - 9 - $2000$
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE0BB060 -> player, jointeam 3 1)
[NoCheatZ v4-beta-r1-158-g01d897e-dirty-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE0BB060 -> player, spec_mode 4)

Program received signal SIGSEGV, Segmentation fault.
0xec7f108f in VisCache::IsValid (this=0x9006740, pa=0x8330ee0, pb=0x8d58e1c)
    at Code/Systems/Blockers/WallhackBlocker.h:104
104                     return m_cache[ pa->GetIndex () ][ pb->GetIndex () ].m_valid;
(gdb) bt
#0  0xec7f108f in VisCache::IsValid (this=0x9006740, pa=0x8330ee0, pb=0x8d58e1c)
    at Code/Systems/Blockers/WallhackBlocker.h:104
#1  0xec7ef8e2 in WallhackBlocker::RT_SetTransmitCallback (this=0x9003ab0, sender_player=..., receiver_player=...)
    at Code/Systems/Blockers/WallhackBlocker.cpp:156
#2  0xec7e42f9 in SetTransmitHookListener::RT_nSetTransmit (This=0xbd79fe0, pInfo=0xebfe7c0, bAlways=false)
    at Code/Hooks/SetTransmitHookListener.cpp:70
#3  0xf2f81ee0 in CServerGameEnts::CheckTransmit(CCheckTransmitInfo*, unsigned short const*, int) ()
   from /mnt/p/home/steam/csgo/csgo/bin/server.so
#4  0xf69b771b in SV_ComputeClientPacks(int, CGameClient**, CFrameSnapshot*) ()
   from /mnt/p/home/steam/csgo/bin/engine.so
#5  0xf69b402e in CGameServer::SendClientMessages(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#6  0xf69b43b1 in SV_SendClientUpdates(bool, bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#7  0xf69b44e0 in SV_Frame(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#8  0xf692308e in _Host_RunFrame_Server(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#9  0xf6923d2c in _Host_RunFrame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#10 0xf6924468 in Host_RunFrame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#11 0xf692d714 in CHostState::State_Run(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#12 0xf692d9e6 in CHostState::FrameUpdate(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#13 0xf692daf9 in HostState_Frame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#14 0xf69ca66a in CEngine::Frame() () from /mnt/p/home/steam/csgo/bin/engine.so
#15 0xf69c75f6 in CDedicatedServerAPI::RunFrame() () from /mnt/p/home/steam/csgo/bin/engine.so
#16 0xf784cc2d in RunServerIteration(bool) () from ./bin/dedicated.so
#17 0xf784ccb8 in RunServer(bool) () from ./bin/dedicated.so
#18 0xf69c76ed in CModAppSystemGroup::Main() () from /mnt/p/home/steam/csgo/bin/engine.so
#19 0xf6b17e70 in CAppSystemGroup::Run() () from /mnt/p/home/steam/csgo/bin/engine.so
#20 0xf69c88b7 in CDedicatedServerAPI::ModInit(ModInfo_t&) () from /mnt/p/home/steam/csgo/bin/engine.so
#21 0xf784cd94 in CDedicatedAppSystemGroup::Main() () from ./bin/dedicated.so
#22 0xf7870740 in CAppSystemGroup::Run() () from ./bin/dedicated.so
#23 0xf787196e in CSteamApplication::Main() () from ./bin/dedicated.so
#24 0xf7870740 in CAppSystemGroup::Run() () from ./bin/dedicated.so
#25 0xf7812d73 in main () from ./bin/dedicated.so
#26 0x080486f9 in main ()

**Comment by L-EARN** [comment]
Have a test with the latest version please.

**Comment by askovpen** [comment]
```
Client "sheriffMONSTER" connected (94.41.126.98:27005).
[NoCheatZ v4-beta-r1-177-g8d61215-debug] DEBUG : CNoCheatZPlugin::ClientConnect (AllowConnect: Yes, EE2BB050 -> , sheriffMONSTER, 94.41.126.98:27005, Connection rejected by game
[NoCheatZ v4-beta-r1-177-g8d61215-debug] DEBUG : CNoCheatZPlugin::ClientPutInServer ( -> sheriffMONSTER) (Was already connected: Yes)
[NoCheatZ v4-beta-r1-177-g8d61215-debug] DEBUG : CNoCheatZPlugin::ClientActive (EE2BB050 -> player)
PutClientInServer: no info_player_start on level
[NoCheatZ v4-beta-r1-177-g8d61215-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, VModEnable 1)
[NoCheatZ v4-beta-r1-177-g8d61215-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, vban 0 0)
[NoCheatZ v4-beta-r1-177-g8d61215-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, joingame )
[NoCheatZ v4-beta-r1-177-g8d61215-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, jointeam 0 1)
L 07/05/2016 - 20:27:48: [csgo_country_v2.smx] 2 round end, 1:1(3) - 9 - $2000$
[NoCheatZ v4-beta-r1-177-g8d61215-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, spec_scoreboard )
[NoCheatZ v4-beta-r1-177-g8d61215-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, spec_scoreboard )
[NoCheatZ v4-beta-r1-177-g8d61215-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, spec_mode 4)
srcds_linux: ../server-plugin/Code/Players/NczPlayerManager.h:270: PlayerHandler::iterator::operator bool() const: Assertion `IsIteratorValid ()' failed.

Program received signal SIGABRT, Aborted.
0xf7fda9f2 in ?? () from /lib/ld-linux.so.2
(gdb) bt
#0  0xf7fda9f2 in ?? () from /lib/ld-linux.so.2
#1  0xf7dc6977 in raise () from /lib32/libc.so.6
#2  0xf7dc7f49 in abort () from /lib32/libc.so.6
#3  0xf7dbfa27 in ?? () from /lib32/libc.so.6
#4  0xf7dbfaab in __assert_fail () from /lib32/libc.so.6
#5  0xec7dc8e1 in PlayerHandler::iterator::operator bool (this=0xffffa004)
    at ../server-plugin/Code/Players/NczPlayerManager.h:270
#6  0xec7e8944 in WallhackBlocker::RT_SetTransmitCallback (this=0x9003ab0, sender_player=..., receiver_player=...)
    at Code/Systems/Blockers/WallhackBlocker.cpp:154
#7  0xec7dc5f9 in SetTransmitHookListener::RT_nSetTransmit (This=0xb333400, pInfo=0xe89b028, bAlways=false)
    at Code/Hooks/SetTransmitHookListener.cpp:73
#8  0xf2f81ee0 in CServerGameEnts::CheckTransmit(CCheckTransmitInfo*, unsigned short const*, int) ()
   from /mnt/p/home/steam/csgo/csgo/bin/server.so
#9  0xf69b771b in SV_ComputeClientPacks(int, CGameClient**, CFrameSnapshot*) ()
   from /mnt/p/home/steam/csgo/bin/engine.so
#10 0xf69b402e in CGameServer::SendClientMessages(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#11 0xf69b43b1 in SV_SendClientUpdates(bool, bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#12 0xf69b44e0 in SV_Frame(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#13 0xf692308e in _Host_RunFrame_Server(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#14 0xf6923d2c in _Host_RunFrame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#15 0xf6924468 in Host_RunFrame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#16 0xf692d714 in CHostState::State_Run(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#17 0xf692d9e6 in CHostState::FrameUpdate(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#18 0xf692daf9 in HostState_Frame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#19 0xf69ca66a in CEngine::Frame() () from /mnt/p/home/steam/csgo/bin/engine.so
#20 0xf69c75f6 in CDedicatedServerAPI::RunFrame() () from /mnt/p/home/steam/csgo/bin/engine.so
#21 0xf784cc2d in RunServerIteration(bool) () from ./bin/dedicated.so
#22 0xf784ccb8 in RunServer(bool) () from ./bin/dedicated.so
#23 0xf69c76ed in CModAppSystemGroup::Main() () from /mnt/p/home/steam/csgo/bin/engine.so
#24 0xf6b17e70 in CAppSystemGroup::Run() () from /mnt/p/home/steam/csgo/bin/engine.so
#25 0xf69c88b7 in CDedicatedServerAPI::ModInit(ModInfo_t&) () from /mnt/p/home/steam/csgo/bin/engine.so
#26 0xf784cd94 in CDedicatedAppSystemGroup::Main() () from ./bin/dedicated.so
#27 0xf7870740 in CAppSystemGroup::Run() () from ./bin/dedicated.so
#28 0xf787196e in CSteamApplication::Main() () from ./bin/dedicated.so
#29 0xf7870740 in CAppSystemGroup::Run() () from ./bin/dedicated.so
#30 0xf7812d73 in main () from ./bin/dedicated.so
#31 0x080486f9 in main ()
```
after flash grenade i have red screen permanent :-). after round end - crash.

**Comment by L-EARN** [comment]
I will try to fix all of this this week.

**Comment by askovpen** [comment]
flash grenade get permanent blind while not round end.

**Comment by askovpen** [comment]
false positive convar detector and after reconnect crash
```
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, +lookatweapon 16)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, -lookatweapon 16)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, +lookatweapon 16)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, -lookatweapon 16)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, +lookatweapon 16)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, -lookatweapon 16)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientDisconnect
Dropped sasai from server: Kicked by Console : [NoCheatZ 4] ConVar request timed out
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, +lookatweapon 16)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, -lookatweapon 16)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, +lookatweapon 16)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, -lookatweapon 16)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, +lookatweapon 16)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB050 -> player, -lookatweapon 16)
Client "sasai" connected (192.168.1.246:27005).
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientConnect (AllowConnect: Yes, EE2BB040 -> , sasai, 192.168.1.246:27005, Connection rejected by game
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientPutInServer ( -> sasai) (Was already connected: Yes)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientActive (EE2BB040 -> player)
PutClientInServer: no info_player_start on level
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, VModEnable 1)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, vban 0 0)
Testing ConCommand say of ПАХОМ
Bypassed ConCommand say of ПАХОМ
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, joingame )
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, jointeam 0 1)
L 07/05/2016 - 20:58:41: [csgo_country_v2.smx] 13 round end, 13:0(3) - 8 - $2000$
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, spec_scoreboard )
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, spec_prev)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, spec_next)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, spec_prev)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, spec_next)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, spec_prev)
[NoCheatZ v4-beta-r1-178-g8da196a-debug] DEBUG : CNoCheatZPlugin::ClientCommand(EE2BB040 -> player, spec_mode 4)
srcds_linux: ../server-plugin/Code/Players/NczPlayerManager.h:270: PlayerHandler::iterator::operator bool() const: Assertion `IsIteratorValid ()' failed.

Program received signal SIGABRT, Aborted.
0xf7fda9f2 in ?? () from /lib/ld-linux.so.2
(gdb) bt
#0  0xf7fda9f2 in ?? () from /lib/ld-linux.so.2
#1  0xf7dc6977 in raise () from /lib32/libc.so.6
#2  0xf7dc7f49 in abort () from /lib32/libc.so.6
#3  0xf7dbfa27 in ?? () from /lib32/libc.so.6
#4  0xf7dbfaab in __assert_fail () from /lib32/libc.so.6
#5  0xec7edebb in PlayerHandler::iterator::operator bool (this=0xffffa018)
    at ../server-plugin/Code/Players/NczPlayerManager.h:270
#6  0xec7f91f6 in WallhackBlocker::RT_SetTransmitCallback (this=0x9003ab0, sender_player=..., receiver_player=...)
    at Code/Systems/Blockers/WallhackBlocker.cpp:154
#7  0xec7edbdb in SetTransmitHookListener::RT_nSetTransmit (This=0xc2fe1d8, pInfo=0xae0c040, bAlways=false)
    at Code/Hooks/SetTransmitHookListener.cpp:73
#8  0xf2f81ee0 in CServerGameEnts::CheckTransmit(CCheckTransmitInfo*, unsigned short const*, int) ()
   from /mnt/p/home/steam/csgo/csgo/bin/server.so
#9  0xf69b771b in SV_ComputeClientPacks(int, CGameClient**, CFrameSnapshot*) ()
   from /mnt/p/home/steam/csgo/bin/engine.so
#10 0xf69b402e in CGameServer::SendClientMessages(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#11 0xf69b43b1 in SV_SendClientUpdates(bool, bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#12 0xf69b44e0 in SV_Frame(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#13 0xf692308e in _Host_RunFrame_Server(bool) () from /mnt/p/home/steam/csgo/bin/engine.so
#14 0xf6923d2c in _Host_RunFrame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#15 0xf6924468 in Host_RunFrame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#16 0xf692d714 in CHostState::State_Run(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#17 0xf692d9e6 in CHostState::FrameUpdate(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#18 0xf692daf9 in HostState_Frame(float) () from /mnt/p/home/steam/csgo/bin/engine.so
#19 0xf69ca66a in CEngine::Frame() () from /mnt/p/home/steam/csgo/bin/engine.so
#20 0xf69c75f6 in CDedicatedServerAPI::RunFrame() () from /mnt/p/home/steam/csgo/bin/engine.so
#21 0xf784cc2d in RunServerIteration(bool) () from ./bin/dedicated.so
#22 0xf784ccb8 in RunServer(bool) () from ./bin/dedicated.so
#23 0xf69c76ed in CModAppSystemGroup::Main() () from /mnt/p/home/steam/csgo/bin/engine.so
#24 0xf6b17e70 in CAppSystemGroup::Run() () from /mnt/p/home/steam/csgo/bin/engine.so
#25 0xf69c88b7 in CDedicatedServerAPI::ModInit(ModInfo_t&) () from /mnt/p/home/steam/csgo/bin/engine.so
#26 0xf784cd94 in CDedicatedAppSystemGroup::Main() () from ./bin/dedicated.so
#27 0xf7870740 in CAppSystemGroup::Run() () from ./bin/dedicated.so
#28 0xf787196e in CSteamApplication::Main() () from ./bin/dedicated.so
#29 0xf7870740 in CAppSystemGroup::Run() () from ./bin/dedicated.so
#30 0xf7812d73 in main () from ./bin/dedicated.so
#31 0x080486f9 in main ()
(gdb)

**Comment by askovpen** [comment]
https://www.youtube.com/watch?v=V6NDMeMQ0DY&feature=youtu.be

**Comment by L-EARN** [comment]
Flashbang fixed (it seems). have a test. (in debug the color from plugin is red it's normal, it's to help compare)

**Comment by L-EARN** [comment]
Fixed in d7330b3
